docs: :memo: document CRDs technical choices (#1321)
containous/traefik-helm-chart: README.md

Accordingly, the encouraged approach to fulfill your needs:

1. Override the default Traefik configuration
   values ([yaml file or cli](https://helm.sh/docs/chart_template_guide/values_files/))
2. Append your own configurations (`kubectl apply -f myconf.yaml`)

[Examples](https://github.com/traefik/traefik-helm-chart/blob/master/EXAMPLES.md) of common usage are provided.

If needed, one may use [extraObjects](./traefik/tests/values/extra.yaml) or extend this

> HelmChart [as a Subchart](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/).

Reviewer: ```suggestion
Helm Chart [as a Subchart](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/).
```

#### With additional CRDs chart

> <details>

Reviewer: Wdyt of putting this bloc in an INFO bloc, like the warning below, something like:

```
> [!INFO]
> <details>....
```

Author: `INFO` doesn't exist, let's use `NOTE` instead.

  <summary>Few words on technical choices about CRDs implementation</summary>

> Some Traefik Helm chart users asked us for help in managing CRDs installed by this chart (cf. #1141, #1209).

Reviewer: ```suggestion
Some Traefik Helm chart users asked us for help in managing CRDs installed by this chart (cf. [#1141](https://github.com/traefik/traefik-helm-chart/issues/1141), [#1209](https://github.com/traefik/traefik-helm-chart/issues/1209)).
```

Helm doesn't support CRDs upgrades (cf. [HIP-0011](https://github.com/helm/community/blob/main/hips/hip-0011.md) for details). 

The objectives are the following:

> 1. continue to support the nominal installation case

Reviewer: ```suggestion
1. Support the nominal installation case following official Helm GuideLines
```

> 2. stay conservative about CRDs to protect resource removal (that's actually one of the reasons why helm doesn't support

Reviewer: ```suggestion
2. Stay conservative about CRDs to protect resource removal (see [HIP 11](https://github.com/helm/community/blob/main/hips/hip-0011.md) for more info)
```

   CRDs upgrades)

> 3. allow users to install multiple instances of Traefik chart along with managed CRDs

Reviewer: ```suggestion
3. Allow users to install multiple instances of Traefik chart along with helm managed CRDs
```
